### PR TITLE
Translation

### PR DESCRIPTION
**Pull request by maribedran**
This PR adds both English and Portuguese versions of the blog post explaining the project setup.

I closed the other one since this is the more complete version of the text.

While translating it I thought the part where I extract the React component into a container and component dirs is not necessary to the post since it's goal is to explain the environment setup and not the framework itself. I'd like your thoughts on  it.

I also want to add another section on image files since it's very similar to the CSS setup and would make a more complete setting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,552 @@
...
+    'DEFAULT': {
+        'BUNDLE_DIR_NAME': 'dist/',
+        'STATS_FILE': os.path.join(BASE_DIR, 'webpack-stats.json'),
+    }
+}
+```
+
+Quanto tivermos terminado as configurações do *front-end*, o projeto terá a seguinte estrutura:
+
+```
```
Comment (thulioph) [inline README.md]: A small visual improvement:

```
├── node_modules
├── project
├── package.json
├── templates
├── babelrc
├── manage.py
├── package.json
├── requirements.txt
├── webpack-stats.json
├── webpack.config.js
└── assets
    └── dist
    └── src
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,552 @@
...
+-requirements.txt
+-webpack-stats.json
+webpack.config.js
+```
+
+Tanto `dist` quanto `webpack-stats.json` serão gerados pelo Webpack.
+
+## Criando o template para a aplicação do *front*
+
+Nosso usuário, quanto acessar o endereço do site, vai fazer isso na raíz do domínio e é nesse ponto que vamos disponibilizar a aplicação React. Para isso, vamos adicionar uma entrada à lista de rotas da aplicação com o *template* que vai receber os arquivos disponibilizados pelo Webpack.
```
Comment (thulioph) [inline README.md]: `quanto`  or `quando`?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,552 @@
...
+from django.views.generic import TemplateView
+from django.urls import path
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('', TemplateView.as_view(template_name='front-end/index.html')),
+]
+```
+
+Normalmente criaríamos uma view dento de um arquivo `views.py` de uma *app*, mas como ela tem somente o propósito de renderizar um *template* sem nenhum objeto de contexto ou lógica associada, apensar adicioná-la à rota deixa explícito que ela não está associada a nenhuma app Django.
```
Comment (thulioph) [inline README.md]: `apensar` or `apenas`?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,552 @@
...
+
+## Iniciando o projeto do *front*
+
+Para começar uma aplicação em React, um dos processos mais simples é usar o [create-react-app](https://github.com/facebook/create-react-app), que já instala e configura todas as bibliotecas necessárias para a aplicação. No entanto, a estrutura gerada por ele tem diversos elementos que não seriam necessários para o nosso caso. Além disso, ele esconde por trás de uma interface muito simples o que está sendo feito. Uma das grandes dificuldades para quem desenvolve originalmente para o *back-end* é entender o papel das diversas bibliotecas da *stack* de desenvolvimento Javascript, por isto escolhemos não utilizar este recurso.
+
+Vamos começar inicializando um projeto com o **NPM** (Node Package Manager), o gerenciador de dependências para ambientes Javascript. Ele se encarrega de baixar, instalar e gerenciar versões de pacotes JS. Se você não tem o NPM no instalado e tem dúvidas sobre como fazer isso, pode conferir a [documentação](https://nodejs.org/en/) ou, buscando no Google, seguir um tutorial com as instruções de instalação para diversas plataformas.
```
Comment (thulioph) [inline README.md]: removes `no` for **no instalado**

File: README.md [hunk 5]
```diff
@@ -1,2 +1,552 @@
...
+
+### [Passo 4](https://github.com/labcodes/django-react-webpack/releases/tag/4)
+
+```
+$ npm init
```
Comment (thulioph) [inline README.md]: or `npm init -y` (yes for all)

File: README.md [hunk 6]
```diff
@@ -1,2 +1,552 @@
...
+...
+```
+
+Agora, para compilar uma única vez os estáticos, podemos rodar no termnal o comando `npm run start`. Já para compilar em tempo real, atualizando o resultado de acordo com cada mudança nos arquivos, utilizaremos o `npm run watch`.
+
+Se observar a árvore de diretórios do projeto, deve ter aparecido um novo chamado `dist` com um arquivo no formato `main-315df4b2f5abf7716c38.js` dentro de `assets`, além do arquivo `webpack-stats.json` na raíz do projeto.
+
+Como o "watch" e o servidor do Django rodando já podemos abrir o endereço no browser e verificar que a aplicação está no ar.
+
+![Servidor e watch rodando](images/img1.png)
```
Comment (thulioph) [inline README.md]: This image is not displayed.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,552 @@
...
+```
+
+Repare que a intalação destes pacotes recebeu um parâmetro diferentes das outros até agora: em vez de `--save-dev` usamos somente `--save`. Isto porque, ao contrário do Webpack, que é uma dependência de desenvolvimento, o código do React vai ser compilado e disponibilizado no Javascript final entregue ao cliente.
+
+Uma característica do React que difere ele de outros *frameworks* como Angular ou JQuery é o *virtual DOM*: ao invés de manipular os elementos já existentes no DOM, ele se encarrega de injetar todo o HTML que aparece para o usuário no elemento raíz da aplicação. É por isto que o template que criamos no Django possui apenas uma `div`, que receberá todo o conteúdo manipulado pelo React. Assim, ao invés de criar arquivos separados de `.html` e `.js`, os componentes do React são na verdade arquivos Javascript que já contêm o HTML que vão renderizar. Para isso, usa-se uma sintaxe especial chamada [JSX](https://reactjs.org/docs/introducing-jsx.html) que depois precisar ser transpilada para virar strings que podem ser lidas pelo interpretador do browser. Quem faz este trabalho de transpilar as sintexes especiais que vamos usar é o [Babel](https://babeljs.io/).
```
Comment (thulioph) [inline README.md]: check the typo for **depois precisar** or `depois precisa`?

File: README.EN.md [hunk 8]
```diff
@@ -0,0 +1,549 @@
+# Django + React + Webpack
+
+Django is one of the most complete web frameworks available on the market. It manages everything from the database to the final HTML sent to the client. But with the introduction of the concept of [*single page apps*](wikipedia) to the front-end development environment, it has become usual to find systems using Django only to provide an [API](wikipedia) on the back-end that responds JSON data to be consumed by Javascript applications. This architecture that divides front and back-end into two separate code bases enables the independence between these sectors on the development process. It also makes it possible to have multiple client apps interacting with one API, ensuring the consistence of data and business rules and at the same time providing different user interfaces.
```
Comment (renatooliveira) [inline README.EN.md]: consistence -> consistency
https://english.stackexchange.com/questions/40564/consistence-and-consistency

File: README.EN.md [hunk 9]
```diff
@@ -0,0 +1,549 @@
+# Django + React + Webpack
+
+Django is one of the most re web frameworks available on the market. It manages everything from the database to the final HTML sent to the client. But with the introduction of the concept of [*single page apps*](wikipedia) to the front-end development environment, it has become usual to find systems using Django only to provide an [API](wikipedia) on the back-end that responds JSON data to be consumed by Javascript applications. This architecture that divides front and back-end into two separate code bases enables the independence between these sectors on the development process. It also makes it possible to have multiple client apps interacting with one API, ensuring the consistency of data and business rules and at the same time providing different user interfaces.
```
Comment (renatooliveira) [inline README.EN.md]: "Django is one of the most *re* web frameworks..."

File: README.EN.md [hunk 10]
```diff
@@ -0,0 +1,549 @@
...
+
+The first part of this post is intended to show how to do the necessary setup for Django to provide the front-end application through its resources for serving static files. After that, we're going to build the basic structure of a React application focusing on showing the roles of each element in it. You won't find here a detailed explanation of how React works, only some core concepts to understand the configuration. The purpose of this post is to empower someone with more background on the back-end to be able to integrate with the front-end more easily without getting lost in the development environment setup. From that point, you can search for other tutorials more specific to the framework itself or even read the documentation, which is very detailed and understandable.
+
+## Django and static files
+
+On a project architecture using Django's built-in [*template engine*](https://docs.djangoproject.com/en/dev/ref/templates/) the static files used on the templates, like `.css` and images, are stored in a directory specified on the `STATICFILES_DIRS` settings variable. You only have to define that variable and Django will serve the files for you.
```
Comment (renatooliveira) [inline README.EN.md]: The staticfiles can be also stored into apps static folders, and those folders they doesn't need to be specified on the STATICFILES_DIRS variable.

File: README.EN.md [hunk 11]
```diff
@@ -0,0 +1,549 @@
...
+    <img src="/static/my_app/example.jpg" alt="My image">
+    ```
+
+What we're going to show on the next part of this tutorial is how to configure a Django project to use a specific template tag that inserts on the HTML the paths to scripts and style sheet files created by Webpack. This page will hold the whole Javascript application, developed on a separate folder inside the Django project.
+
+## Configuring the Django project
+
+### [Step 1](https://github.com/labcodes/django-react-webpack/releases/tag/1)
+
+The initial setup we need is just a standard ]Django installation](https://docs.djangoproject.com/en/dev/intro/tutorial01/), optionally with [REST Framework](http://www.django-rest-framework.org/#installation) for the building the API, and the [Django Webpack Loader](https://github.com/ezhome/django-webpack-loader) package.
```
Comment (renatooliveira) [inline README.EN.md]: the `]Django installation]` tag is wrong

**Pull request merged**